base/linalg/triangular.jl

```diff
@@ -1547,12 +1547,11 @@ for (f, g) in ((:A_mul_Bc, :A_mul_Bc!), (:A_mul_Bt, :A_mul_Bt!))
     end
 end
 
-for mat in (:AbstractVector, :AbstractMatrix)
-
 ### Multiplication with triangle to the left and hence rhs cannot be transposed.
 for (f, g) in ((:*, :A_mul_B!), (:Ac_mul_B, :Ac_mul_B!), (:At_mul_B, :At_mul_B!))
     @eval begin
-        function ($f)(A::AbstractTriangular, B::$mat)
+        ($f)(A::AbstractTriangular, B::AbstractVector) = ($f)(A, reshape(B, Val{2}))
```

Comment:
If I read this correctly, I believe this changes the behavior such that the rhs is now promoted to a matrix, right?